server/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorServiceTests.java

```diff
@@ -393,10 +429,27 @@ private static ClusterState createClusterStateWith(List<IndexRoutingTable> index
                     )
                 )
         );
-
+        Metadata.Builder metadataBuilder = Metadata.builder();
+        if (indexNameToPriorityMap != null) {
```

Comment:
This map enables a lot of things to happen in the cluster state (ie. an entire IndexMetadata to be created). 

What do you think if `index` is renamed to `indexRoutingTable` and we add another method to create `indexMetadata` (which receives the index name and priority for now)? Then `createClusterStateWith` will receives a list of IndexRoutingTable and a list of IndexMetadata?

Reply:
I think it's rather unexpected that passing a priority configuration creates the IndexMetadata with all the necessary settings. That's why I suggested creating the IndexMetadatas explicitly (with the corresponding priorities) and passing them to the `createClusterStateWith` method

Reply:
I'm not sure I'm crazy about the change -- it complicates things a little for the calling code -- but I've made it.